app/misc/graphing.ts

```diff
@@ -453,7 +517,37 @@ function makeAbsNode(c, column: number) {
                 });
 
                 abEdges.add({
-                    from: id + numOfCommits * (i + 1),
+                    from: bsnodeId,
+                    to: id
+                });
+            }
+        }
+
+        // Initializing viewable tags in second zoom graph level
```

Comment:
I would suggest indicating what the difference is in this algorithm vs the previous. I understand one is zoomed out. It looks very similar (I do see differences, but perhaps these can be extracted out?)

Reply:
Any suggestions? I am having the hardest time defining the middle ground zoom when it's so similar to the highest zoom level in that the commits are still grouped in a seemingly random way.

Reply:
I was thinking you could extract this functionality into a helper function, since it is mostly copy/pasted code, something like:
function anAppropriateName(c, tags, nodes, edges) {
    if (c.toString() in tags) {
        for (let i = 0; i < tags[c.toString()].length; i++) {
            let tagName = tags[c.toString()][i];
            let tp = tagName.name().split("/");
            let shortTagName = tp[tp.length - 1];
            console.log(shortTagName + " tag: " + tagName.isHead().toString());
            if (tagName.isHead()) {
                shortTagName = "*" + shortTagName;
            }
            let bsnodeId = generateUniqueNumber();
            nodes.add({
                id: bsnodeId,
                shape: "ellipse",
                // color: "teal",
                title: tagName, // hover text
                label: shortTagName, // shown under/in shape
                physics: false,
                fixed: false,
                x: (column - 0.6 * (i + 1)) * tagSpacingX,
                y: (id - 0.3) * tagSpacingY,
            });

            edges.add({
                from: bsnodeId,
                to: id
            });

            tagIds[tagid] = bsnodeId;
        }
    }
}

Does that make sense?